core-bundle/src/EventListener/Widget/CustomRgxpListener.php

```diff
@@ -35,7 +36,7 @@ public function __invoke(string $regexp, mixed $input, Widget $widget): bool
             return true;
         }
 
-        if (!preg_match($widget->customRgxp, $input)) {
+        if (!preg_match($widget->customRgxp, StringUtil::decodeEntities($input))) {
```

Comment:
The above confusion comes from the fact that I missed that you used `StringUtil::decodeEntities` not as I meant. What I meant in https://github.com/contao/contao/issues/7322#issuecomment-2210429017 was the following:

```suggestion
        if (!preg_match(StringUtil::decodeEntities($widget->customRgxp), $input)) {
```

Reply:
Ok I did not thought of that way. But by doing so, it will also not work as `$input`is always encoded.

Reply:
>  But by doing so, it will also not work as $inputis always encoded.

That depends on the widget.